Update supported data type table

MKKnorr opened the pull request:
16 bit integers are supported. Removed the note about some tests not passing, as this does not belong in the documentation, and should be fixed instead

MKKnorr committed:
Update supported data type table

Use the type names from the precision landing page of the rocm documentation

TorreZuk commented:
IMO failing tests should be documented in a known bugs or limitations section until they are fixed.  There is no need for a user to find an existing known bug and report it as it is not documented.

neon60 commented on docs/data-type-support.rst:
```diff
@@ -21,45 +21,56 @@ Data type support
       - ✅
-
-* The ⚠️ means that the data type is mostly supported, but there are some API tests, that do not work.
-   * The ``block_histogram`` test fails with ``int8``.
```
Let's keep to notes here.

MKKnorr commented:
> IMO failing tests should be documented in a known bugs or limitations section until they are fixed. There is no need for a user to find an existing known bug and report it as it is not documented.

I agree, but this does not belong in the documentation, as that state changes over time. I added it to the "known issues" section in the changelog

Rmalavally commented:
> 16 bit integers are supported. Removed the note about some tests not passing, as this does not belong in the documentation, and should be fixed instead

Is there a way to track and update these known issues when fixed? Is the component owner aware of the known issue documentation?

MKKnorr commented:
> > 16 bit integers are supported. Removed the note about some tests not passing, as this does not belong in the documentation, and should be fixed instead
> 
> Is there a way to track and update these known issues when fixed? Is the component owner aware of the known issue documentation?

I opened internal issues for the rocprim team and informed them about it

Rmalavally commented:
> > > 16 bit integers are supported. Removed the note about some tests not passing, as this does not belong in the documentation, and should be fixed instead
> > 
> > 
> > Is there a way to track and update these known issues when fixed? Is the component owner aware of the known issue documentation?
> 
> I opened internal issues for the rocprim team and informed them about it

Awesome. Thanks so much.

neon60 commented:
@stanleytsang-amd please review, and merge in, if it looks good.

MKKnorr committed:
Update Changelog

MKKnorr commented:
Rebased on top of https://github.com/ROCm/rocPRIM/pull/541 as that includes changes that affect this PR. Updated the changelog to reflect that.

https://github.com/ROCm/rocPRIM/pull/541 Needs to be merged first

umfranzw reviewed (approved):
I think the documentation changes look good. However, as mentioned above, we'll need to  wait until [#541](https://github.com/ROCm/rocPRIM/pull/541) is merged before this one goes in.

MKKnorr commented:
@umfranzw thanks, I rebased it, as #541 is now merged. Please merge when convenient
